src/layup/convert.py

```diff
@@ -113,47 +126,67 @@ def _apply_convert(data, convert_to, cache_dir=None, primary_id_column_name=None
 
     if convert_to not in ["BCART", "BCOM", "BKEP", "CART", "COM", "KEP"]:
         raise ValueError("Invalid conversion type")
+    has_covariance = has_cov_columns(data)
+    logger.debug(f"Data does not have covariance: {not has_covariance}")
```

Comment:
Minor point - perhaps flip this to remove the "not"?

```suggestion
    logger.debug(f"Data has covariance: {has_covariance}")
```